walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -11,13 +11,180 @@
+        }
+    }
+    return holderList;
+  }
+ 
+  private boolean checkForOverlap(Collection<String> mandatory, Set<String> attendees){
+      //compares to see if there is overlap in the attendees of a collection and set
+      for(String person:attendees){
+          if(mandatory.contains(person)){
+              return true;
+          }
+      }
+ 
+      return false;
+  }
+
+ 
+  private ArrayList<TimeRange> removeConflict(ArrayList<TimeRange> freeSlots, Event event){
+    TimeRange takenSlot = event.getWhen();
+    int takenSlotStart = takenSlot.start();
+    int takenSlotEnd = takenSlot.end();
+ 
+    ArrayList<TimeRange> timeRangeHolderList = new ArrayList<TimeRange>();
+    ArrayList<Integer> indexHolderList = new ArrayList<Integer>();
+ 
+    for(TimeRange slot: freeSlots){
+        if(slot.start() <= takenSlotStart && slot.end() >= takenSlotEnd){
+          //  |------free--------|
+          //     |--event--|
+          int slotIndex = freeSlots.indexOf(slot);
+          TimeRange priorSlotTime = TimeRange.fromStartEnd(slot.start(),takenSlotStart,false);
+          TimeRange postSlotTime = TimeRange.fromStartEnd(takenSlotEnd,slot.end(),false);
+ 
+          indexHolderList.add(slotIndex);
+          timeRangeHolderList.add(postSlotTime);
+          timeRangeHolderList.add(priorSlotTime);
+        }else if(slot.start() < takenSlotStart && slot.end() > takenSlotStart){
+          // |-----free----|
+          //      |----event---|
+          int slotIndex = freeSlots.indexOf(slot);
+          TimeRange priorSlotTime = TimeRange.fromStartEnd(slot.start(),takenSlotStart,false);
+ 
+          indexHolderList.add(slotIndex);
+          timeRangeHolderList.add(null);//holder in the array
+          timeRangeHolderList.add(priorSlotTime);
+        }else if(slot.start() < takenSlotEnd && slot.end() > takenSlotEnd){
+          //      |----free-----|
+          // |---event----|
+          int slotIndex = freeSlots.indexOf(slot);
+          TimeRange postSlotTime = TimeRange.fromStartEnd(takenSlotEnd,slot.end(),false);
+ 
+          indexHolderList.add(slotIndex);
+          timeRangeHolderList.add(null);//holder in the array
+          timeRangeHolderList.add(postSlotTime);
+        }
+    }
+    
+    int j = 0;
+    int holderLength = timeRangeHolderList.size();
+    for(int i=0;i<holderLength;i+=2){
+        int slotIndex = indexHolderList.get(i);
+        freeSlots.remove(slotIndex);
+
+        if(holderLength > i && timeRangeHolderList.get(i) != null){
+        TimeRange postSlot = timeRangeHolderList.get(i);
+        freeSlots.add(slotIndex,postSlot);
+        }
+
+        if(holderLength > i+1 && timeRangeHolderList.get(i+1) != null){
+            TimeRange priorSlot = timeRangeHolderList.get(i+1);
+            freeSlots.add(slotIndex,priorSlot);
+        }
+        j++;
+    }
+    return freeSlots;
+  }
+
+ 
+  private ArrayList<TimeRange> cleanFreeSlots(ArrayList<TimeRange> freeSlots, long duration){
```

Comment:
nit: method name could be more descriptive, for example: `removeSlotsBelowThreshold`